core/src/env/engine/on_chain/impls.rs

```diff
@@ -330,6 +330,21 @@ impl TypedEnv for EnvInstance {
         ext::restore_to(account_id, code_hash, rent_allowance, filtered_keys);
     }
 
+    fn transfer<T>(&mut self, destination: T::AccountId, value: T::Balance) -> Result<()>
+    where
+        T: EnvTypes,
+    {
+        // Reset the contract-side buffer to append onto clean slate.
+        self.reset_buffer();
+        // Append the encoded `account_id`, `code_hash` and `rent_allowance`
+        // and `filtered_keys` in order and remember their encoded regions
+        // within the buffer.
```

Comment:
See CI for compiler errors in this method, also this comment needs to be updated. Looks like a copy paste error.